### PR TITLE
Support custom labels for pulsar zk, and bookie cluster

### PR DESCRIPTION
**Pull request by tuteng**
* Support custom labels for pulsar zk, and bookie cluster

### CODE REVIEW

File: charts/sn-platform/values.yaml [hunk 1]
```diff
@@ -746,6 +746,7 @@ zookeeper:
   ## Retention Policy
   resourcePolicy:
     keep: false
+  labels: []
```
Comment (fantapsody) [inline charts/sn-platform/values.yaml:749]: Should the type of labels be a map (`{}`)?

File: charts/sn-platform/templates/zookeeper/zookeeper-cluster.yaml [hunk 2]
```diff
@@ -29,13 +29,19 @@ metadata:
...
+    {{ toYaml . | indent 4 }}
+    {{- end }}
 spec:
   replicas: {{ .Values.zookeeper.replicaCount }}
   image: "{{ .Values.images.zookeeper.repository }}:{{ .Values.images.zookeeper.tag }}"
   pod:
     labels:
       {{- include "pulsar.template.labels" . | nindent 6 }}
       component: {{ .Values.zookeeper.component }}
+      {{- with .Values.broker.labels }}
```
Comment (fantapsody) [inline charts/sn-platform/templates/zookeeper/zookeeper-cluster.yaml:42]:
```suggestion
      {{- with .Values.zookeeper.labels }}
```

File: charts/sn-platform/templates/proxy/proxy-cluster.yaml [hunk 3]
```diff
@@ -38,6 +41,9 @@ spec:
     labels:
       {{- include "pulsar.template.labels" . | nindent 6 }}
       component: {{ .Values.proxy.component }}
+      {{- with .Values.broker.labels }}
```
Comment (fantapsody) [inline charts/sn-platform/templates/proxy/proxy-cluster.yaml:44]:
```suggestion
      {{- with .Values.proxy.labels }}
```

File: charts/sn-platform/templates/proxy/proxy-cluster.yaml [hunk 4]
```diff
@@ -29,6 +29,9 @@ metadata:
   labels:
     {{- include "pulsar.standardLabels" . | nindent 4 }}
     component: {{ .Values.proxy.component }}
+    {{- with .Values.broker.labels }}
```
Comment (fantapsody) [inline charts/sn-platform/templates/proxy/proxy-cluster.yaml:32]:
```suggestion
    {{- with .Values.proxy.labels }}
```

File: charts/sn-platform/templates/bookkeeper/bookkeeper-cluster.yaml [hunk 5]
```diff
@@ -37,6 +40,9 @@ spec:
     labels:
       {{- include "pulsar.template.labels" . | nindent 6 }}
       component: {{ .Values.bookkeeper.component }}
+      {{- with .Values.broker.labels }}
```
Comment (fantapsody) [inline charts/sn-platform/templates/bookkeeper/bookkeeper-cluster.yaml:43]:
```suggestion
      {{- with .Values.bookkeeper.labels }}
```

File: charts/sn-platform/templates/bookkeeper/bookkeeper-cluster.yaml [hunk 6]
```diff
@@ -28,6 +28,9 @@ metadata:
   labels:
     {{- include "pulsar.standardLabels" . | nindent 4 }}
     component: {{ .Values.bookkeeper.component }}
+    {{- with .Values.broker.labels }}
```
Comment (fantapsody) [inline charts/sn-platform/templates/bookkeeper/bookkeeper-cluster.yaml:31]:
```suggestion
    {{- with .Values.bookkeeper.labels }}
```

**Comment by tuteng** [comment]
Thanks @fantapsody, I've fixed your comment, another question, how can I add a custom label and selector to the service?

File: charts/sn-platform/templates/bookkeeper/bookkeeper-cluster.yaml [hunk 7]
```diff
@@ -28,6 +28,9 @@ metadata:
   labels:
     {{- include "pulsar.standardLabels" . | nindent 4 }}
     component: {{ .Values.bookkeeper.component }}
+    {{- with .Values.bookkeeper.labels }}
+    {{ toYaml . | indent 4 }}
```
Comment (MarvinCai) [inline charts/sn-platform/templates/bookkeeper/bookkeeper-cluster.yaml]: I think we shouldn't indent the line here, the indent function already did the work

Comment (MarvinCai) [inline charts/sn-platform/templates/bookkeeper/bookkeeper-cluster.yaml]:
```suggestion
{{ toYaml . | indent 4 }}
```

Comment (MarvinCai) [reply charts/sn-platform/templates/bookkeeper/bookkeeper-cluster.yaml]: same for other places where we include yaml

Comment (tuteng) [reply charts/sn-platform/templates/bookkeeper/bookkeeper-cluster.yaml]: Thanks, fixed

Comment (tuteng) [reply charts/sn-platform/templates/bookkeeper/bookkeeper-cluster.yaml]: Thanks, fixed

**Review by fantapsody — APPROVED** [review]

**Review by MarvinCai — APPROVED** [review]

**Pull request merged**